cpp/src/slice2java/Gen.cpp

```diff
@@ -1854,22 +1870,27 @@ Slice::Gen::TypesVisitor::visitClassDefStart(const ClassDefPtr& p)
                 baseDataMembers = baseClass->allDataMembers();
             }
 
-            if (hasRequiredMembers && hasOptionalMembers)
+            if (!requiredMembers.empty() && !optionalMembers.empty())
             {
                 //
                 // Generate a constructor accepting parameters for just the required members.
                 //
-                out << sp << nl << "public " << name << spar;
+                out << sp;
+                out << nl << "/**";
+                out << nl << " * Creates a new {@code " << name
+                    << "} with values for all fields not marked optional in the Slice definition for {@code "
+                    << p->scoped() << "}.";
+                out << nl << " *";
+                writeParamDocComments(out, requiredMembers);
```

Comment:
There's a couple things here:
1) If there are no parameters, we have an extra newline for no reason from the `nl << *`.
We should only add this spacing if there's going to be more tags after.

2) Maybe this is intentional, but `writeParamDocComments` only writes user-provided doc comments. So you can get things like this. So if a user doesn't document their members or something, you get this:
```
    /**
     * Creates a new {@code C} with values for all fields not marked optional in the Slice definition for {@code ::Test::C}.
     *
     */
    public C(int requiredA, int requiredB, String ss)
```
Maybe this is fine, but won't it complain about missing @param tags?

Reply:
Fixed.